tests/test_sorting.py

```diff
@@ -0,0 +1,367 @@
+        6.906, 7.753, 8.757, 9.955, 11.392, 13.222, 15.351, 17.953, 21.157, 25.128,
+        30.082, 36.305, 44.621, 54.768, 67.779, 84.572, 106.378, 134.858, 172.244,
+        224.385, 290.538, 378.283, 494.762, 649.232, 853.428],
+    exptime_emgain_sec=[5, 10, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 
+        5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 10, 10, 10,
+        10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10],
+    nframes=[3, 5, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
+        3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 5, 5, 5,
+        5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5]
+        ):
+    """
+    Create an array of CMDGAIN, EXPTIME for frames that will be used to
+    calibrate EM-gain vs DAC.
+
+     Args:
+      em_gain (list): set of ditinct (non-unity) EM gains chosen to collect
+        data for EM gain with pupil images
+      exptime_emgain_sec (list): set of distinct exposure times in seconds chosen to
+        collect frames for non-linearity calibration for each EM gain
+      nframes (int): number of frames per ditinct exposure time
+
+    Returns:
+      cmdgain_list (list): list of commanded gains
+      exptime_list (list): list of exposure frames
+    """
+    # Create pairs of frames
+    nonunity_gain_list = []
+    for idx in range(len(em_emgain)):
+        nonunity_gain_list += [em_emgain[idx]] * nframes[idx] 
+    cmdgain_list = [1] * np.sum(nframes) + nonunity_gain_list
+    exptime_list = []
+    for idx in range(len(exptime_emgain_sec)):
+        exptime_list += [exptime_emgain_sec[idx]] * nframes[idx]
+    # Unity and non-unity gains
+    exptime_list += exptime_list
+    return cmdgain_list, exptime_list
+
+def make_minimal_image(
+    cmdgain=1,
+    exptime_sec=0,
+    frameid=0,
+        ):
+    """
+    This function makes a mock frame with minimum memory in its data and error
+    fields. It is used in this test script only.
+
+    Args:
+      cmdgain (float): commanded gain of the frame
+      exptime_sec (float): exposure time of the frame
+      frameid (int): an integer value used to indentify the frame
+
+    Returns:
+      filename (String): filename with path of the generated FITS file
+    """
+    signal = np.zeros(1)
+
+    prhd, exthd = create_default_headers()
+    # Mock error maps
+    err = np.ones(1)
+    dq = np.zeros(1, dtype = np.uint16)
+    # Creating a FITS file to assign it a filename with the frame ID
+    prim = fits.PrimaryHDU(header = prhd)
+    hdr_img = fits.ImageHDU(signal, header=exthd)
+    hdul = fits.HDUList([prim, hdr_img])
+    # Record actual commanded EM
+    hdul[1].header['CMDGAIN'] = cmdgain
+    # Record actual exposure time
+    hdul[1].header['EXPTIME'] = exptime_sec
+    filename = str(Path('simdata', f'CGI_EXCAM_L1_{frameid:0{10}d}.fits'))
+    hdul.writeto(filename, overwrite = True)
+    return filename
+
+# Main code
```

Comment:
A unit test needs to be inside a function starting with "test_". It can't just be in a script like this.

Reply:
Done